frappe/core/doctype/doctype/doctype.js

```diff
@@ -217,5 +207,11 @@ frappe.ui.form.on("DocField", {
 			$doctype_select.val(curr_value.doctype);
 			update_fieldname_options();
 		}
+	},
+
+	fieldtype: function(frm) {
+		frm.trigger("max_attachments");
 	}
 });
+
+extend_cscript(cur_frm.cscript, new frappe.model.DocTypeController({frm: cur_frm}));
```

Comment:
@surajshetty3416 I am little bit skeptical of this change. Not sure what will be the side effects.

Reply:
I don't think there will be any side effects with this...  `extend_cscript` itself takes care of one of the issues.